core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js

```diff
@@ -834,24 +803,24 @@ oppia.directive('conversationSkin', [
                     }
                     FatigueDetectionService.reset();
                     NumberAttemptsService.reset();
-                    _nextFocusLabel = FocusManagerService.generateFocusLabel();
+                    _nextFocusLabel = focusLabel;
 
-                    PlayerTranscriptService.setDestination(newStateName);
+                    PlayerTranscriptService.setDestination(
+                      nextCard.getStateName());
 
                     // These are used to compute the dimensions for the
                     // next card.
-                    $scope.upcomingStateName = newStateName;
-                    $scope.upcomingParams = newParams;
-                    $scope.upcomingContentHtml = (
-                      contentHtml + _getRandomSuffix());
+                    $scope.nextCard = nextCard;
+                    $scope.nextCard.addRandomSuffixToContentHtml(
+                      _getRandomSuffix());
+                    $scope.nextCard.addRandomSuffixToInteractionHtml(
+                      _getRandomSuffix());
 
                     var _isNextInteractionInline = (
                       ExplorationPlayerStateService.isNextInteractionInline());
```

Comment:
This seems odd, can't you get the info from nextCard?

Reply:
Yeah, it could be, but that would require importing INTERACTION_SPECS completely into StateCardObjectFactory. Is that fine?

Reply:
Yes that's fine. Constants can be imported anywhere.

Reply:
Done.